manifests/init.pp

```diff
@@ -133,6 +134,13 @@
   validate_string($computers_ou)
   validate_string($nismaps_ou)
 
+  if is_string($domain_change) {
+    $domain_change_real = str2bool($domain_change)
+  } else {
+    $domain_change_real = $domain_change
+  }
+  validate_bool($domain_change_real)
```

Comment:
Stringified bool handling should also be tested. You could re-use my newest incarnation of validation tests in https://github.com/ghoneycutt/puppet-module-ssh/blob/v3.34.0/spec/classes/init_spec.rb#L3482-L3511